lsqr_spot.m

```diff
@@ -234,7 +234,12 @@
    Nv = (1/alfa)*Nv;
 end
 
-Arnorm = alfa*beta;     if Arnorm == 0, disp(msg(1,:)); return, end
+Arnorm = alfa*beta;     
+
+if Arnorm == 0
+    % Bypass main loop
+    itn = itnlim;
+end
```

Comment:
That's not good because `itn` is collected in the stats at the end, and it's not the actual number of iterations. It would be better to introduce a boolean that will skip over the main loop. Something like `optimal = Arnorm <= ...` consistently with the tests at the end of the loop.